scripts/sign_image.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash -ex
+
+
+IMAGE=""
+SIGNING_KEY=""
+SIGNING_CERT=""
+CA_CERT=""
+TARGET_PATH=""
+TMP_CERT_PATH="/tmp/cert"
+CERT_PATH=""
+
+usage()
+{
+    echo "Usage:  $0 -i <image_path> [-c <cert_path>] [-t <cert_target_path>]"
+    exit 1
+}
+
+generate_signing_key()
+{
+    SIGNING_CSR="${TMP_CERT_PATH}/signing.csr"
+    CA_KEY="${TMP_CERT_PATH}/ca.key"
+    mkdir -p "${TMP_CERT_PATH}"
+
+    # Generate the CA key and certificate
+    openssl genrsa -out $CA_KEY 4096
+    openssl req -x509 -new -nodes -key $CA_KEY -sha256 -days 3650 -subj "/C=US/ST=Seattle/L=Redmond/O=SONiC/CN=www.sonic.com" -out $CA_CERT
+
+    # Generate the signing key, certificate request and certificate
+    openssl genrsa -out $SIGNING_KEY 4096
+    openssl req -new -key $SIGNING_KEY -subj "/C=US/ST=Seattle/L=Redmond/O=SONiC/CN=www.sonic.com" -out $SIGNING_CSR
+    openssl x509 -req -in $SIGNING_CSR -CA $CA_CERT -CAkey $CA_KEY -CAcreateserial -out $SIGNING_CERT -days 1825 -sha256
+
+    # Remove no use files
+    rm -f $CA_CEY
+    rm -f $SIGNING_CSR
+}
+
+while getopts ":i:c:t:" opt; do
+    case $opt in
+        i)
+            IMAGE=$OPTARG
+            ;;
+        c)
+            CERT_PATH=$OPTARG
```

Comment:
>CERT_PATH=$OPTARG [](start = 12, length = 17)

Design the CLI with
- a parameter for SIGNING_CERT path
- a parameter for CA_CERT path
- a parameter for SIGNING_KEY path

And the behaviors:
- if all provided, sign as requested
- if non provided, test sign
- if partially provided, error

All other internal files should be `mktemp`

Reply:
Changed the command line.